clang/lib/Basic/DiagnosticIDs.cpp

```diff
@@ -509,30 +499,32 @@ DiagnosticIDs::getDiagnosticLevel(unsigned DiagID, SourceLocation Loc,
 diag::Severity
 DiagnosticIDs::getDiagnosticSeverity(unsigned DiagID, SourceLocation Loc,
                                      const DiagnosticsEngine &Diag) const {
-  assert(getBuiltinDiagClass(DiagID) != CLASS_NOTE);
+  const bool IsCustomDiag = DiagnosticIDs::IsCustomDiag(DiagID);
```

Comment:
We don't typically use top-level const on locals (here and elsewhere in the patch).